### PR TITLE
chore(cli): prompt to select app if not under a workspace

### LINKED ISSUE #1238: env init returns error if not in a workspace
When I run `copilot env init` and I'm not in a copilot workspace, then instead of prompting me to choose which application do I want to use, the CLI just errors out saying `✘ no application found: run `app init` or `cd` into your workspace please`.

### PR DESCRIPTION
**Pull request by Lou1415926**
This pull requests changes current cli behavior for `env init`. If the user is not currently in any workspace and no application is specified using `--app` flag, it prompts the user to select from existing applications instead of returning an error. 

Resolves #1238  

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/pkg/cli/env_init.go [hunk 1]
```diff
@@ -192,6 +195,12 @@ func (o *initEnvOpts) Validate() error {
 
 // Ask asks for fields that are required but not passed in.
 func (o *initEnvOpts) Ask() error {
+	if o.appName == "" {
```
Comment (efekarakus) [inline internal/pkg/cli/env_init.go]: nit: to be consistent with the other `ask*` methods can we move this if-check within `askAppName`?

File: internal/pkg/cli/env_init.go [hunk 2]
```diff
@@ -32,6 +32,8 @@ import (
 )
 
 const (
+	envInitAppNameHelpPrompt = "An environment will be created to the selected application."
```
Comment (efekarakus) [inline internal/pkg/cli/env_init.go]:
```suggestion
	envInitAppNameHelpPrompt = "An environment will be created in the selected application."
```

File: internal/pkg/cli/env_init.go [hunk 3]
```diff
@@ -66,6 +68,7 @@ https://aws.github.io/copilot-cli/docs/credentials/#environment-credentials`
 )
 
 var (
+	envInitAppNamePrompt = fmt.Sprintf("In which %s would you like to create an environment?", color.Emphasize("application"))
```
Comment (efekarakus) [inline internal/pkg/cli/env_init.go]:
```suggestion
	envInitAppNamePrompt = fmt.Sprintf("In which %s would you like to create the environment?", color.Emphasize("application"))
```

**Review by efekarakus — COMMENTED** [review]
Looks great! I am no English expert :) so maybe someone else should double check my suggestions but looks good to me otherwise

**Review by huanjani — APPROVED** [review]
Looks great!

**Review by efekarakus — APPROVED** [review]

**Pull request merged**